mypy/type_visitor.py

```diff
@@ -347,3 +349,121 @@ def query_types(self, types: Iterable[Type]) -> T:
                 self.seen_aliases.add(t)
             res.append(t.accept(self))
         return self.strategy(res)
+
+
+class TypeQueryBool(SyntheticTypeVisitor[bool]):
+    """Specialized visitor for boolean strategies
+
+    strategy is used to combine results for a series of types,
+    Cases involve a boolean query using `any` or `all`.
+
+    Note: this visitor keeps an internal state (tracks type aliases to avoid
+    recursion), so it should *never* be re-used for querying different types,
+    create a new visitor instance instead.
+
+    # TODO: check that we don't have existing violations of this rule.
+    """
+    STRATEGY_ANY = 0  # type: Final[int]
+    STRATEGY_ALL = 1  # type: Final[int]
+
+    def __init__(self, strategy: int) -> None:
+        # 0: any()
```

Comment:
Drop this comment, since the constants have names

Reply:
✅